test/irmin-pack/test_pack_version_bump.ml

```diff
@@ -0,0 +1,169 @@
+
+module Log = (val Logs.src_log src : Logs.LOG)
+
+module Util = struct
+
+  (** Following are generic utils *)
+
+  let tmp_dir () = Filename.temp_file "test_pack_version_bump_" ""
+
+  (** Make a directory; dir is assumed to not exist; parent is assumed to exist *)
+  let mkdir p =
+    assert(not (Sys.file_exists p));
+    Filename.quote_command "mkdir" [p] |> fun cmd ->
+    [%log.info "exec: %s\n%!" cmd];
+    Sys.command cmd |> function
+    | 0 -> ()
+    | n ->
+      Fmt.failwith "Command `%s' exited with non-zero code %d\n" cmd n
+
+
+  (** Copy src to dst; dst is assumed to not exist *)
+  let copy_dir src dst =
+    assert(not (Sys.file_exists dst));
+    (* don't check if it is empty; perhaps we should *)
+    Filename.quote_command "cp" ["-R";src;dst] |> fun cmd ->
+    [%log.info "exec: %s\n%!" cmd];
+    Sys.command cmd |> function
+    | 0 -> ()
+    | n ->
+      Fmt.failwith "Failed to set up the test env; command `%s' exited \
+                    with non-zero code %d\n" cmd n
+
+  (** More specific utils from here *)
+
+  let v1_store_archive_dir = "test/irmin-pack/data/version_1"
+
+  (** Check we are in the location we expect, the project root. See
+     test_existing_stores.goto_project_root for alternative. *)
+  let assert_project_root () =
+    match Sys.file_exists v1_store_archive_dir with
+    | true -> ()
+    | false ->
+      Fmt.failwith "Couldn't find dir %S from cwd %S\n%!"
+        v1_store_archive_dir
+        Filename.current_dir_name
+
+  let _ = assert_project_root ()
+
+  module Unix_ = Irmin_pack.IO.Unix
+
+  (** Get the version of the underlying file; file is assumed to
+     exist; file is assumed to be an Irmin_pack.IO.Unix file *)
+  let io_get_version ~fn : [`V1 | `V2] =
+    assert(Sys.file_exists fn);
+    Unix_.v ~version:None ~fresh:false ~readonly:true fn |> fun t ->
+    let r = Unix_.version t in
+    Unix_.close t;
+    r
+
+end
+open Util
+
+
+(** This sets up infrastructure to open the existing "version_1" store *)
+module With_existing_store() = struct
+
+  let tmp_dir = tmp_dir ()
+
+  let _ = [%log.info "Using temporary directory %s\n%!" tmp_dir]
+
+  (* Make a copy of the v1_store_archive_dir in tmp_dir *)
+  let _ =
+    assert_project_root();
+    rm_dir tmp_dir;
+    copy_dir v1_store_archive_dir tmp_dir;
+    ()
+
+  (** Set up modules to allow access to "version_1" store *)
+  module Store_ = struct
+    (* In test_existing_stores.ml, we have Test_reconstruct, which
+       uses S.traverse_pack_file; so S is likely the correct config
+       for the pack file. S = V2(), V2() = V2_maker.Make(Schema);
+       V2_maker = Irmin_pack.Maker(Conf); Conf is from Tezos_irmin;
+       Schema is from common *)
+    module Conf = Irmin_tezos.Conf
+
+    module Schema = Common.Schema
+    (* same as Irmin_tezos.Schema? no; tezos uses their own hash;
+       common has Hash.SHA1 *)
+
+    (* from test_existing_stores.ml; FIXME why V2? *)
```

Comment:
No good reason; `test_existing_stores` just defines two different configs (separated by differing inode parameters). The extraction of this code to a separate module would be a good time to give them better names...